### PR TITLE
Dear ImGui Implementation

### PR DESCRIPTION
**Pull request by viniciuslrangel**
Press F10 to show FPS 😄 

- [x] Add core headers
- [x] Adapt SDL3 & Vulkan backend
- [x] Integrate to driver pipeline
- [x] Basic input handling (keyboard/mouse)
- [x] Gamepad capture
- [x] Layer abstraction
- [x] Basic video information layer (frame timing)

### CODE REVIEW

**Comment by viniciuslrangel** [comment]
~Dear ImGui is not as a submodule yet because it needs patches to compile with C++23~

Forked and added as submodule [ext-imgui](https://github.com/shadps4-emu/ext-imgui)

File: .gitmodules [hunk 1]
```diff
@@ -81,4 +81,9 @@
 [submodule "externals/ffmpeg-core"]
 	path = externals/ffmpeg-core
 	url = https://github.com/shadps4-emu/ext-ffmpeg-core.git
-	shallow = true
\ No newline at end of file
+	shallow = true
+[submodule "externals/dear_imgui"]
+	path = externals/dear_imgui
+	url = git@github.com:shadps4-emu/ext-imgui.git
```
Comment (sampletext32) [inline .gitmodules]: all neighboring urls are `https`. 
```suggestion
	url = https://github.com/shadps4-emu/ext-imgui.git
```

File: .gitmodules [hunk 2]
```diff
@@ -81,4 +81,9 @@
 [submodule "externals/ffmpeg-core"]
 	path = externals/ffmpeg-core
 	url = https://github.com/shadps4-emu/ext-ffmpeg-core.git
-	shallow = true
\ No newline at end of file
+	shallow = true
+[submodule "externals/dear_imgui"]
+	path = externals/dear_imgui
```
Comment (abouvier) [inline .gitmodules:89]: Why not just `imgui`?

File: externals/CMakeLists.txt [hunk 3]
```diff
@@ -149,6 +149,19 @@ if (APPLE AND NOT TARGET date::date-tz)
     add_subdirectory(date)
 endif()
 
+# Dear ImGui
+if (NOT TARGET Dear_ImGui)
```
Comment (abouvier) [inline externals/CMakeLists.txt]: The target will never exists if there is no `find_package` ;-)

File: .gitmodules (hunk 2 above)
Comment (viniciuslrangel) [reply .gitmodules:89]: I'll forward the declaration later

File: externals/CMakeLists.txt [hunk 4]
```diff
@@ -155,6 +155,19 @@ if (APPLE)
     endif()
 endif()
 
+# Dear ImGui
+block()
```
Comment (abouvier) [inline externals/CMakeLists.txt]: [block()](https://cmake.org/cmake/help/latest/command/block.html) was added in CMake 3.25 but shadps4 uses `cmake_minimum_required(VERSION 3.16.3)`.

Comment (viniciuslrangel) [reply externals/CMakeLists.txt]: Oh, thanks for the tip. Will fix later

**Comment by gandalfthewhite19890404** [comment]
It is for QT builds only?

**Comment by viniciuslrangel** [comment]
> It is for QT builds only?

No, both!

**Comment by viniciuslrangel** [comment]
Removed fix commits & ensured stability in each

**Review by raphaelthegreat — APPROVED** [review]
Looks awesome, thanks a lot for this work!

**Comment by Digote** [comment]
When I try to press here, this error occurs and the emulator closes.

image

**Comment by viniciuslrangel** [comment]
> When I try to press here, this error occurs and the emulator closes.
> 
> image
> 

Can you create a issue with more details, please

**Pull request merged**